src/main/java/gyro/google/compute/InstanceGroupManagerResource.java

```diff
@@ -0,0 +1,166 @@
+
+        for (String changedFieldName : changedFieldNames) {
+            // template changes
+            if (changedFieldName.equals("instance-template")) {
+                setInstanceTemplate();
+            } else if (changedFieldName.equals("target-size")) {
+                instanceGroupManager.setTargetSize(getTargetSize());
+                shouldPatch = true;
+            } else if (changedFieldName.equals("auto-healing-policy")) {
+                List<ComputeInstanceGroupManagerAutoHealingPolicy> diffableAutoHealingPolicy = getAutoHealingPolicy();
+                List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicies = null;
+
+                if (diffableAutoHealingPolicy.isEmpty()) {
+                    autoHealingPolicies = Data.nullOf(List.class);
+                } else {
+                    autoHealingPolicies = diffableAutoHealingPolicy
+                        .stream()
+                        .map(ComputeInstanceGroupManagerAutoHealingPolicy::copyTo)
+                        .collect(Collectors.toList());
+                }
+                instanceGroupManager.setAutoHealingPolicies(autoHealingPolicies);
+                shouldPatch = true;
+            } else if (changedFieldName.equals("named-port")) {
+                // TODO: investigate as GCP UI allows updating of named ports.
+            }
+        }
+
+        if (shouldPatch) {
+            patch(instanceGroupManager);
+        }
+    }
+
+    @Override
+    public void doDelete(GyroUI ui, State state) throws Exception {
+        Compute client = createClient(Compute.class);
+        Operation operation = client.instanceGroupManagers()
+            .delete(getProjectId(), getZone(), getName())
+            .execute();
+        waitForCompletion(client, operation);
+    }
+
+    @Override
+    public void copyFrom(InstanceGroupManager model) {
+        super.copyFrom(model);
+
+        // Do NOT update zone with a full url as this should be a name.
+        //        setZone(model.getZone());
+    }
+
+    private void setInstanceTemplate() throws Exception {
+        Compute client = createClient(Compute.class);
+        InstanceGroupManagersSetInstanceTemplateRequest request = new InstanceGroupManagersSetInstanceTemplateRequest();
+        request.setInstanceTemplate(getInstanceTemplate().getSelfLink());
+        Operation operation = client.instanceGroupManagers()
+            .setInstanceTemplate(getProjectId(), getZone(), getName(), request)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    private void patch(InstanceGroupManager instanceGroupManager) throws Exception {
+        Compute client = createClient(Compute.class);
+        Operation operation = client.instanceGroupManagers()
+            .patch(getProjectId(), getZone(), getName(), instanceGroupManager)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
```

Comment:
Refresh should be called at the end the create/update unless you need to reset a value before.